### PR TITLE
[implementations] Refactor code, improve docstrings

### PR DESCRIPTION
**Pull request by aswanipranjal**
@Polaris000, @jgbarah, @valeriocos

This PR introduces some improvements on the docstrings and refactors some of the code in the implementations directory. 

Please have a look and let me know if you agree or want me to change anything, thanks!

### CODE REVIEW

**Comment by Polaris000** [comment]
Thanks @aswanipranjal. Please have a look at #174. @jgbarah has made some significant changes. A couple of the changes you suggest are there in #175 as well.

File: implementations/code_df/code_changes_git.py [hunk 1]
```diff
@@ -67,21 +68,21 @@ def compute_timeseries(self, period='month'):
...
                 .groupby([df['created_date'].dt.year.rename('year'),
-                          df['created_date'].dt.week.rename('week')])   \
+                          df['created_date'].dt.week.rename('week')]) \
                 .agg('count')
 
             all_periods = pd.DataFrame(
-                            pd.date_range(self.since, self.until, freq='W'),
-                            columns=['Dates'])
+                pd.date_range(self.since, self.until, freq='W'),
+                columns=['Dates'])
```
Comment (Polaris000) [inline implementations/code_df/code_changes_git.py]: I had run the code through a `flake8` test. Is there a specific reason you suggest this change? Thanks

File: implementations/code_df/commit.py [hunk 2]
```diff
@@ -40,50 +40,49 @@ def __init__(self, items, date_range=(None, None), issrccode_obj=None):
...
+    def _flatten_data(self, item):
         """
         This method is for cleaning a raw commit fetched by Perceval.
         Since all attributes of the data are not of importance, it is
         better to just keep the ones which are required.
 
-        :param line: a raw line fetched by Perceval, present in the JSON file
-            It is a dictionary.
+        :param item: a raw item (of type dict) fetched by Perceval,
+        present in the JSON file (items object)
```
Comment (Polaris000) [inline implementations/code_df/commit.py]: I had indented all following lines after the `param` tag. 
Like this:
```python
:param xyz: line1
    line2
    line3..
```
Is that a bad practise? Just curious.

File: implementations/code_df/code_changes_git.py (hunk 1 above)
Comment (aswanipranjal) [reply implementations/code_df/code_changes_git.py]: These have happened when I ran the formatter in pyCharm, @Polaris000

File: implementations/code_df/commit.py (hunk 2 above)
Comment (aswanipranjal) [reply implementations/code_df/commit.py]: It is not a bad practice, @Polaris000. This is the result of the formatter as well.

File: implementations/code_df/code_changes_git.py (hunk 1 above)
Comment (Polaris000) [reply implementations/code_df/code_changes_git.py]: OK Thanks!

**Comment by aswanipranjal** [comment]
@Polaris000, [#174](https://github.com/chaoss/wg-evolution/pull/174) changes the README.md file. Did you mean to point me towards [#177](https://github.com/chaoss/wg-evolution/pull/177) instead?

I actually didn't see it before, my bad.

**Comment by Polaris000** [comment]
Oops, sorry. Yeah I meant to point you to #177.

**Comment by jgbarah** [comment]
@aswanipranjal thanks a lot for this pr. After merging #177 I'm not sure if is better to rebase this one, to open a new one, or what... What do you think?

**Comment by aswanipranjal** [comment]
@jgbarah, I'll rebase with master and open a new PR with the remaining changes (if any)

**Pull request closed without merging**